plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -962,6 +965,34 @@ static void registerOrgQbiccObjectModelIntrinsics(final CompilationContext ctxt)
 
         };
         intrinsics.registerIntrinsic(Phase.LOWER, objModDesc, "set_initialized", typeIdVoidDesc, set_initialized);
+
+        FieldElement nativeObjectMonitorField = CoreClasses.get(ctxt).getObjectNativeObjectMonitorField();
+        // public static native PThread.pthread_mutex_t_ptr nativeObjectMonitor_of(Object reference);
+        MethodDescriptor nomOfDesc = MethodDescriptor.synthesize(classContext, pthreadMutexDesc, List.of(objDesc));
+        StaticIntrinsic nomOf = (builder, target, arguments) -> {
+            Value mutexSlot = builder.load(builder.instanceFieldOf(builder.referenceHandle(arguments.get(0)), nativeObjectMonitorField), MemoryAtomicityMode.NONE);
+            PointerType returnType = (PointerType)target.getType().getReturnType();
+            return builder.valueConvert(mutexSlot, returnType);
+        };
+        intrinsics.registerIntrinsic(Phase.LOWER, objModDesc, "nativeObjectMonitor_of", nomOfDesc, nomOf);
+
+        // public static native Stdint.uint8_t set_nativeObjectMonitor(Object object, PThread.pthread_mutex_t_ptr nom);
```

Comment:
Why not return a java boolean here?

Reply:
It does now. This comment is out of date.